lib/plugins/aws/package/compile/events/stream/index.js

```diff
@@ -11,40 +11,83 @@ class AwsCompileStreamEvents {
     };
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8034
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'stream', {
-      anyOf: [{ type: 'string' }, { type: 'object' }],
+      anyOf: [
+        { $ref: '#/definitions/awsArnString' },
+        {
+          type: 'object',
+          properties: {
+            arn: { $ref: '#/definitions/awsArn' },
+            type: { enum: ['dynamodb', 'kinesis'] },
+            batchSize: { type: 'integer', minimum: 1, maximum: 10000 },
+            parallelizationFactor: { type: 'integer', minimum: 1, maximum: 10 },
+            startingPosition: { enum: ['LATEST', 'TRIM_HORIZON'] },
+            enabled: { type: 'boolean' },
+            consumer: { oneOf: [{ const: true }, { $ref: '#/definitions/awsArn' }] },
+            batchWindow: { type: 'integer', minimum: 0, maximum: 300 },
+            maximumRetryAttempts: { type: 'integer', minimum: -1, maximum: 10000 },
+            bisectBatchOnFunctionError: { type: 'boolean' },
+            maximumRecordAgeInSeconds: {
+              oneOf: [
+                { type: 'integer', minimum: -1, maximum: -1 },
+                { type: 'integer', minimum: 60, maximum: 604800 },
+              ],
+            },
+            destinations: {
+              type: 'object',
+              properties: {
+                onFailure: {
+                  anyOf: [
+                    { $ref: '#/definitions/awsArnString' },
+                    {
+                      type: 'object',
+                      properties: {
+                        arn: { $ref: '#/definitions/awsArn' },
+                        type: { enum: ['sns', 'sqs'] },
+                      },
+                      additionalProperties: false,
+                      required: ['arn'],
+                      oneOf: [
+                        {
+                          properties: {
+                            arn: { $ref: '#/definitions/awsCfFunction' },
+                          },
+                          required: ['type'],
+                        },
+                        {
+                          properties: {
+                            arn: { $ref: '#/definitions/awsArnString' },
+                          },
+                        },
+                      ],
+                    },
+                  ],
+                },
+              },
+              additionalProperties: false,
+              required: ['onFailure'],
+            },
+          },
+          additionalProperties: false,
+          required: ['arn'],
+          oneOf: [
+            {
+              properties: {
+                arn: { $ref: '#/definitions/awsCfFunction' },
+              },
+              required: ['type'],
+            },
+            {
+              properties: {
+                arn: { $ref: '#/definitions/awsArnString' },
+              },
+            },
+          ],
```

Comment:
I guess similarly here, `arn` and `type` definition could be handled purely in scope of `oneOf`